apiVersion: openapi/v1alpha1
kind: Radar
metadata:
    name: "26886954"
    labels:
        datastore_id: "6174817271152640"
data:
    classification: Other Bug
    created: "2016-06-20T06:35:56.87349Z"
    description: "Summary:\r\nWhen migrating our codebase from Swift 2.2 -> Swift 3.0 we noticed that when we had multiple chained methods that acquired `throws` in the migration, `try!` was added twice.\r\n\r\nSteps to Reproduce:\r\n1. Define an expression where multiple functions are called that now have `throws` in Swift 3.0. Example:\r\n\r\n```\r\nlet remoteFileURL = baseAssetsURL\r\n    .URLByAppendingPathComponent(remoteFileName + fileSuffix)\r\n    .URLByAppendingPathExtension(\"zip\")\r\n```\r\n\r\n2. Migrate the code\r\n\r\nExpected Results:\r\n`try!` is added a single time, resulting in:\r\n\r\n```\r\nlet remoteFileURL = try! baseAssetsURL\r\n    .appendingPathComponent(remoteFileName + fileSuffix)\r\n    .appendingPathExtension(\"zip\")\r\n```\r\n\r\n\r\nActual Results:\r\nInstead, two `try!`s are added, resulting in:\r\n\r\n```\r\nlet remoteFileURL = try! try! baseAssetsURL\r\n    .appendingPathComponent(remoteFileName + fileSuffix)\r\n    .appendingPathExtension(\"zip\")\r\n```\r\n\r\nVersion:\r\nXcode 8 beta 1\r\n\r\nNotes:"
    email: keithbsmiley@gmail.com
    modified: "2016-07-06T06:12:02.03811Z"
    number: "26886954"
    number_intvalue: 26886954
    originated: 19-Jun-2016 23:35
    parent_number: '&{NULL_VALUE}'
    product: Developer Tools
    product_version: Xcode 8 beta 1
    reproducible: Always
    resolved: ""
    status: Duplicate of 26924050 (open)
    title: Swift 3.0 migrator adds try twice if methods are chained
